### PR TITLE
Add 18-741 elective

### PR DESCRIPTION
**Pull request by tanayasija**
# Summary of Changes

- Please indicate the classes you are modifying:
  - 18-741

# Checklist:

- [x] I have added my entries to the `README.md` table of contents in the root directory as links.

Make sure that your contributions:

- [x] Are accurate and will not mislead the reader
- [x] Do not encourage or cause academic integrity violations
- [x] Do not offend professors or unfairly criticize the university

### CODE REVIEW

File: electives/18741.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+This class is a great introduction to Computer Networks covering all the 
+layers of the network stack with the physical and data-link layer covered in the first half
+of the course and network, transport and application layer covered in the second half
+of the course. The course does not assume background with socket programming and dealing
+with highly concurrent programs, however experience with the same through other systems courses
+like 18-213 should be great help. The course is quite comprehensive in providing a whirlwind tour
+of networks and helps in building good intuition and experience in network programming. This class
+used to traditionally be offered in the Spring but has now been shifted to the Fall.
+
+## Language Used
```
Comment (mikinty) [inline electives/18741.md:18]: nit: add space after header (markdown style but not super important)

File: electives/18741.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+the lectures cover the physical layer involving details around modulation, bit encoding, link bandwidth and noise immunity. The next few lectures deep dive into the data-link layer covering concepts around flow control, medium access control, framing and error detection and correction. The second half of the course focuses heaily on the network and transport layer and talks about routing protocols and algorithms, addressing and naming in the Internet. The lectures also cover the two most well known transport protocols including UDP and TCP, their semantics, techniques for congestion control. The last few lectures dive into the application layer protocol and techniques that are transforming the network such as content delivery networks, software defined networks, 5G and other interesting research topics. The recommended textbook for this course is Computer Networks: A Systems Approach by Bruce S. Davie and Larry L. Peterson, online free versions of which are available on the Internet as well as at CMU Libraries. The library also has hard copies of the book available as well.   
+
+## Quiz
```
Comment (mikinty) [inline electives/18741.md:30]: ditto space after header

File: electives/18741.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+the lectures cover the physical layer involving details around modulation, bit encoding, link bandwidth and noise immunity. The next few lectures deep dive into the data-link layer covering concepts around flow control, medium access control, framing and error detection and correction. The second half of the course focuses heaily on the network and transport layer and talks about routing protocols and algorithms, addressing and naming in the Internet. The lectures also cover the two most well known transport protocols including UDP and TCP, their semantics, techniques for congestion control. The last few lectures dive into the application layer protocol and techniques that are transforming the network such as content delivery networks, software defined networks, 5G and other interesting research topics. The recommended textbook for this course is Computer Networks: A Systems Approach by Bruce S. Davie and Larry L. Peterson, online free versions of which are available on the Internet as well as at CMU Libraries. The library also has hard copies of the book available as well.
```
Comment (mikinty) [inline electives/18741.md]: would be nice to add a link to the textbook if you can

File: electives/18741.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+with students implementing their own routing algorithms to get an idea of large scale routing design and challenges. The third project focuses on the transport layer and involves the implementation of delivering a byte stream reliably in the presence of packet loss, re-ordering and duplication. The final project involves designing a content delivery server capable of handling hundereds of concurrent clients. All the projects involve heavy socket programming and dealing with concurrent programs making them challenging and fun at the same time.
+
+## Exams
+
+Exams are not designed to be super tricky or difficult but instead focus on the concepts covered in lectures and in projects. Going through past year paper should help in gaining familiarity with the format,
+number of questions and time management. The exams usually do not test concepts based on memory and a cheatsheet is allowed in order to avoid such memorization and promote understanding the reasons and idea behind concepts well. A few questions can be tricky, but going through the lectures and projects thoroughly should allow one to answer them intuitively.
+
+## How to do well
```
Comment (mikinty) [inline electives/18741.md:44]: any links to pass course websites / exams?

**Review by mikinty — APPROVED** [review]
This is great, thanks for adding!

File: electives/18741.md (hunk 2 above)
Comment (tanayasija) [reply electives/18741.md:30]: Incorporated the change

File: electives/18741.md (hunk 4 above)
Comment (tanayasija) [reply electives/18741.md:44]: Unfortunately no :( Those are released by the professor on canvas

File: electives/18741.md (hunk 3 above)
Comment (tanayasija) [reply electives/18741.md]: Added the link as well

**Pull request merged**